numba/testing.py

```diff
@@ -49,49 +47,15 @@ def run_tests(suite, xmloutput=None, verbosity=1, nomultiproc=False):
         runner = xmlrunner.XMLTestRunner(output=xmloutput)
     else:
         runner = None
-    prog = NumbaTestProgram(suite=suite, testRunner=runner, exit=False,
+    prog = NumbaTestProgram(module=None,
+                            testRunner=runner, exit=False,
                             verbosity=verbosity,
                             nomultiproc=nomultiproc)
     return prog.result
 
 
 def test(**kwargs):
-    """
-    Run all tests under ``numba.tests``.
-
-    kwargs
-    ------
-    - descriptions
-    - verbosity
-    - buffer
-    - failfast
-    - xmloutput [str]
-        Path of XML output directory
-    """
-    from numba import cuda
-
-    suite = discover_tests("numba.tests")
```

Comment:
There are still various 'runtests.py' scripts, notably in the cuda test suites. I'm not sure how relevant these scripts are at this point, i.e. whether anyone would run them explicitly.

This pull request should in fact render them obsolete, so we may remove them as part of a 'cleanup' follow-up patch.